File: WebServer/LVSEmergency/src/main/java/it/lvsemergency/areaInformationManagement/data/AprsData.java
package it.lvsemergency.areaInformationManagement.data;

import javax.persistence.EmbeddedId;
import javax.persistence.Entity;

import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonUnwrapped;

@Table(name = "aprsdata")
@Entity
public class AprsData {
	@JsonUnwrapped
	@EmbeddedId
	private AprsDataId aprsDataId;
	
	private Float temperature;
	private Float pressure;
	private Integer humidity;
	private Float windDirection;
	private Float windSpeed;
	private Float windGust;
	private Float rainOneHour;
	private Float rainDay;
	private Float rainMidNight;
	private Float luminosity;

	public AprsData() {

	}

	public AprsData(AprsDataId aprsDataId, Float temperature, Float pressure, Integer humidity, Float windDirection, Float windSpeed,
			Float windGust, Float rainOneHour, Float rainDay, Float rainMidNight, Float luminosity) {
		super();
		this.aprsDataId = aprsDataId;
		this.temperature = temperature;
		this.pressure = pressure;
		this.humidity = humidity;
		this.windDirection = windDirection;
		this.windSpeed = windSpeed;
		this.windGust = windGust;
		this.rainOneHour = rainOneHour;
		this.rainDay = rainDay;
		this.rainMidNight = rainMidNight;
		this.luminosity = luminosity;
	}

	public AprsDataId getAprsDataId() {
		return aprsDataId;
	}
	
	public void setAprsDataId(AprsDataId aprsDataId) {
		this.aprsDataId = aprsDataId;
	}
	
	public Float getTemperature() {
		return temperature;
	}

	public void setTemperature(Float temperature) {
		this.temperature = temperature;
	}

	public Float getPressure() {
		return pressure;
	}

	public void setPressure(Float pressure) {
		this.pressure = pressure;
	}

	public Integer getHumidity() {
		return humidity;
	}

	public void setHumidity(Integer humidity) {
		this.humidity = humidity;
	}

	public Float getWindDirection() {
		return windDirection;
	}

	public void setWindDirection(Float windDirection) {
		this.windDirection = windDirection;
	}

	public Float getWindSpeed() {
		return windSpeed;
	}

	public void setWindSpeed(Float windSpeed) {
		this.windSpeed = windSpeed;
	}

	public Float getWindGust() {
		return windGust;
	}

	public void setWindGust(Float windGust) {
		this.windGust = windGust;
	}

	public Float getRainOneHour() {
		return rainOneHour;
	}

	public void setRainOneHour(Float rainOneHour) {
		this.rainOneHour = rainOneHour;
	}

	public Float getRainDay() {
		return rainDay;
	}

	public void setRainDay(Float rainDay) {
		this.rainDay = rainDay;
	}

	public Float getRainMidNight() {
		return rainMidNight;
	}

	public void setRainMidNight(Float rainMidNight) {
		this.rainMidNight = rainMidNight;
	}

	public Float getLuminosity() {
		return luminosity;
	}

	public void setLuminosity(Float luminosity) {
		this.luminosity = luminosity;
	}

}

File: ClientApp/PacClientApp/src/areaInformationManagement/aprsdata.cpp
#include "aprsdata.h"

namespace areaInformationManagementIF {
AprsData::AprsData(QObject *parent)
    : helpers::EntityIF{}
{
    setParent(parent);
}

QJsonDocument AprsData::toJsonDocument()
{
    return QJsonDocument::fromJson("");
}

void AprsData::fromJsonObject(QJsonObject jsonObject)
{
    name = jsonObject.value("name").toString();
    date = QDateTime::fromString(jsonObject.value("time").toString(), "dd-MM-yyyy HH:mm:ss");
    temperature = jsonObject.value("temperature").toDouble();
    pressure = jsonObject.value("pressure").toDouble();
    humidity = jsonObject.value("humidity").toDouble();
    windDirection = jsonObject.value("windDirection").toDouble();
    windSpeed = jsonObject.value("windSpeed").toDouble();
    windGust = jsonObject.value("windGust").toDouble();
    rainOneHour = jsonObject.value("rainOneHour").toDouble();
    rainDay = jsonObject.value("rainDay").toDouble();
    rainMidNight = jsonObject.value("rainMidNight").toDouble();
    luminosity = jsonObject.value("luminosity").toDouble();
}

QString AprsData::getName()
{
    return name;
}

void AprsData::setName(QString newName)
{
    name = newName;
}

QString AprsData::getDate()
{
    return date.toString("dd-MM-yyyy HH:mm:ss");
}

void AprsData::setDate(QDateTime newDate)
{
    date = newDate;
}

float AprsData::getTemperature() const
{
    return temperature;
}

void AprsData::setTemperature(float newTemperature)
{
    temperature = newTemperature;
}

float AprsData::getPressure() const
{
    return pressure;
}

void AprsData::setPressure(float newPressure)
{
    pressure = newPressure;
}

int AprsData::getHumidity() const
{
    return humidity;
}

void AprsData::setHumidity(int newHumidity)
{
    humidity = newHumidity;
}

float AprsData::getWindDirection() const
{
    return windDirection;
}

void AprsData::setWindDirection(float newWindDirection)
{
    windDirection = newWindDirection;
}

float AprsData::getWindSpeed() const
{
    return windSpeed;
}

void AprsData::setWindSpeed(float newWindSpeed)
{
    windSpeed = newWindSpeed;
}

float AprsData::getWindGust() const
{
    return windGust;
}

void AprsData::setWindGust(float newWindGust)
{
    windGust = newWindGust;
}

float AprsData::getRainOneHour() const
{
    return rainOneHour;
}

void AprsData::setRainOneHour(float newRainOneHour)
{
    rainOneHour = newRainOneHour;
}

float AprsData::getRainDay() const
{
    return rainDay;
}

void AprsData::setRainDay(float newRainDay)
{
    rainDay = newRainDay;
}

float AprsData::getRainMidNight() const
{
    return rainMidNight;
}

void AprsData::setRainMidNight(float newRainMidNight)
{
    rainMidNight = newRainMidNight;
}

float AprsData::getLuminosity() const
{
    return luminosity;
}

void AprsData::setLuminosity(float newLuminosity)
{
    luminosity = newLuminosity;
}
}
